include/alpaka/vec/Vec.hpp

```diff
@@ -96,9 +96,9 @@ namespace alpaka
         }
 
     private:
-        template<typename F, TVal... Is>
-        ALPAKA_FN_HOST_ACC constexpr explicit Vec(F&& generator, std::integer_sequence<TVal, Is...>)
-            : m_data{generator(std::integral_constant<TVal, Is>{})...}
+        template<typename F, typename TIdx, TIdx... Is, typename = std::enable_if_t<std::is_integral_v<TIdx>>>
+        ALPAKA_FN_HOST_ACC constexpr explicit Vec(F&& generator, std::integer_sequence<TIdx, Is...>)
+            : m_data{generator(std::integral_constant<TIdx, Is>{})...}
```

Comment:
It seems very odd to me to constrain a private API, since we basically control all callsites. This looks rather like a design error. Fortunately, I wrote that code so I remember what I tried to do :)

This constructor supports the one above it and should mimic [`std::simd`'s generator constructor](https://en.cppreference.com/w/cpp/experimental/simd/simd/simd), which uses `std::size_t` for the index sequence. For some reason I used `TVal` here. Probably because my brain was thinking about `Vec`'s of integers only.

The correct fix IMO would be to just use `std::index_sequence` and `std::size_t`:
```c++
#else
        template<typename F, std::enable_if_t<std::is_invocable_v<F, std::integral_constant<std::size_t, 0>>, int> = 0>
        ALPAKA_FN_HOST_ACC constexpr explicit Vec(F&& generator)
#endif
            : Vec(std::forward<F>(generator), std::make_index_sequence<TDim::value>{})
        {
        }

    private:
        template<typename F, std::size_t... Is>
        ALPAKA_FN_HOST_ACC constexpr explicit Vec(F&& generator, std::index_sequence<Is...>)
            : m_data{generator(std::integral_constant<std::size_t, Is>{})...}
        {
        }
```

Reply:
@sliwowitz COuld you have again a look into what @bernhardmgruber wrote?